Hazel/Dtls/DtlsUnityConnection.cs

```diff
@@ -349,25 +352,31 @@ protected override void HandleSend(byte[] data, byte sendOption, Action ackCallb
         }
 
         /// <inheritdoc />
-        protected override void WriteBytesToConnection(byte[] bytes, int length)
+        protected override void WriteBytesToConnection(SmartBuffer bytes, int length)
         {
-            ByteSpan wireData = this.WriteBytesToConnectionInternal(bytes, length);
+            SmartBuffer wireData = this.WriteBytesToConnectionInternal(bytes, length);
+            bytes.Recycle();
```

Comment:
This assumes that something called AddUsage() on the SmartBuffer before this method was called right?

Reply:
I thiiiink in the call sites below, AddUsage() might not be called